src/Microsoft.AspNetCore.Rewrite/Extensions/HttpRequestExtensions.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Rewrite.Extensions
+{
+    public static class HttpRequestExtensions
+    {
+        public static Uri ToUri(this HttpRequest request)
```

Comment:
@Tratcher it's also used in my [global rules PR](https://github.com/aspnet/BasicMiddleware/pull/169/commits/d99270576e9fb996ffec40d52ed74a8d3f623cb6)